opentargets_pharmgkb/evidence_generation.py

```diff
@@ -100,26 +156,46 @@ def chebi_id_to_iri(id_):
     return f'http://purl.obolibrary.org/obo/CHEBI_{id_}'
 
 
+def explode_and_map_phenotypes(df):
```

Comment:
Not a dumb question, these actually aren't present in the ClinVar repo at all.  (You can see what functionality is shared with that repo by checking the `cmat` imports at the top of the files.)  These functions are mostly "glue" that lets us apply that functionality to the PharmGKB, which is a different beast for at least a few reasons:
1. PGKB data is always strictly tabular, as opposed to CV's very rich but also complex XML.  This makes using Pandas dataframes more of a natural fit for PGKB.
2. PGKB has much less data - I don't have detailed numbers, but `wc -l` reveals about 5000 clinical annotations in PGKB as opposed to CV's 3 million RCVs.  This means we can actually read all the data into memory, which makes the code simpler.
3. I'm assuming PGKB will have no manual curation, which along with (2) means it's more reasonable to run all the mappings, external API calls, etc. in a single Python script, rather than asynchronously and managed via Nextflow and manual processing.

For example, for `explode_and_map_phenotypes` the approximately equivalent function in CMAT is [group_diseases_by_efo_mapping](https://github.com/EBIvariation/eva-opentargets/blob/master/cmat/output_generation/clinvar_to_evidence_strings.py#L376) which ends up looking quite different for all these reasons.

Sorry for getting carried away here, but it's a good question and always worth keeping in mind the reasons for these decisions so we can revisit them if the situation changes.

Reply:
Thanks for the elaborated answer!
I get it a bit better now :)